src/BackTranslationHelper/BackTranslationHelperCtrl.cs

```diff
@@ -811,5 +896,137 @@ private void DisplayRighttoleftToolStripMenuItem_DropDownOpening(object sender,
+				return null;
+			}
+
+			return findWhat;
+		}
+
+		private string GetSelectedText()
+		{
+			var findWhat = textBoxTargetBackTranslation.SelectedText;
+			if (String.IsNullOrEmpty(findWhat))
+			{
+				// if there is no selected text in the box, see if they want to use what's on the clipboard
+				IDataObject iData = Clipboard.GetDataObject();
+
+				// Determines whether the data is in a format you can use.
+				if (iData.GetDataPresent(DataFormats.UnicodeText))
+				{
+					findWhat = (string)iData.GetData(DataFormats.UnicodeText);
+				}
+			}
+
+			return findWhat;
+		}
+
+		private void findSubstitutionRuleMenuItem_Click(object sender, EventArgs e)
+		{
+			if (TheFindReplaceProject == null)
+			{
+				if (String.IsNullOrEmpty(CheckInitializeFindReplaceHelper()))
+				{
+					return;
+				}
+				// TheFindReplaceProject should be initialized if we're here
+			}
+
+			string findWhat;
+			if ((findWhat = GetRequiredSelectedText()) == null)
+				return;
+
+			TheFindReplaceProject.FindReplacementRule(findWhat);
+
+			Initialize(!String.IsNullOrEmpty(_model.TargetDataPreExisting) && !hideCurrentTargetTextToolStripMenuItem.Checked);
+			UpdateData(_model);
+		}
+
+		private void editSubtitutionsMenuItem_Click(object sender, EventArgs e)
+		{
+			if (TheFindReplaceProject == null)
+			{
+				if (String.IsNullOrEmpty(CheckInitializeFindReplaceHelper()))
+				{
+					return;
+				}
+				// TheFindReplaceProject should be initialized if we're here
+			}
+
+			TheFindReplaceProject.EditSpellingFixes();
+
+			Initialize(!String.IsNullOrEmpty(_model.TargetDataPreExisting) && !hideCurrentTargetTextToolStripMenuItem.Checked);
+			UpdateData(_model);
+		}
+
+		private void assignNewSubstitutionProjectMenuItem_Click(object sender, EventArgs e)
+		{
+			// check if we're using a FindReplace helpers
+			if (Properties.Settings.Default.MapProjectNameToFindReplaceProject != null)
+			{
+				var mapProjectNameToFindReplaceProjects = SettingToDictionary(Properties.Settings.Default.MapProjectNameToFindReplaceProject);
+				var projectName = BackTranslationHelperDataSource.ProjectName;
+				if (mapProjectNameToFindReplaceProjects.ContainsKey(projectName))
+				{
+					mapProjectNameToFindReplaceProjects.Remove(projectName);
+					Properties.Settings.Default.MapProjectNameToFindReplaceProject = SettingFromDictionary(mapProjectNameToFindReplaceProjects);
+					Properties.Settings.Default.Save();
+				}
+			}
+
+			TheFindReplaceProject = null;
+			ButtonSubstitute_Click(sender, e);
+		}
+
+		private FindReplaceHelper TrySpellFixerProjectLogin(string findWhat)
+		{
+			try
+			{
+				var aSF = FindReplaceHelper.GetFindReplaceHelper();
+				DirectableEncConverter.EncConverters.Reinitialize();
+				aSF.QueryForSpellingCorrectionIfTableEmpty(findWhat);
+				return aSF;
+			}
+			catch (Exception)
+			{
+				// usually just a "no project selected message, so .... ignoring it
```

Comment:
It would be better if the ignored exception was narrower. Is there a specific type thrown if you see "no project selected"?

Reply:
changed it to show the message box in any case _aside_ from the ‘No project selected‘ case, which we don‘t care about.